### ISSUE TITLE
Update CocoaPod dependencies

### ISSUE THREAD

**Issue by ide** [aws/aws-sdk-ios#113]
A few of the dependencies in the podspec have been updated: 
```
$ pod outdated
Analyzing dependencies
The following updates are available:
- Mantle 1.4.1 -> 1.4.1 (latest version 1.5.1)
- Reachability 3.1.1 -> 3.1.1 (latest version 3.2)
- UICKeyChainStore 1.0.5 -> 1.0.5 (latest version 1.1.0)
```

**Comment by ide** [comment]
Would you be open to omitting the patch portion of the version number (as in "major.minor.patch") in the Podfile? For example, the Podfile would say `pod 'Mantle', '~> 1.4'` instead of `pod 'Mantle', '~> 1.4.1'`. This is the format that CocoaPods recommends by default.

This allows users to use Mantle 1.4.x, 1.5.x, etc. but not Mantle 2.x.x, which should be OK as long as Mantle follows the semver spec and always bumps the major version when breaking backwards compatibility.

**Comment by yosuke-matsuda** [comment]
We've released the AWS Mobile SDK for iOS 2.0.14. We made the dependency versions less restrictive with this version.

Thanks,
